src/Gui/MainWindow.cpp

```diff
@@ -1917,13 +1917,17 @@ void MainWindow::startSplasher()
             GetGroup("BaseApp")->GetGroup("Preferences")->GetGroup("General");
         // first search for an external image file
         if (hGrp->GetBool("ShowSplasher", true)) {
-            d->splashscreen = new SplashScreen(this->splashImage());
+            d->splashscreen = new SplashScreen(this->splashImage(),
+                Qt::WindowStaysOnTopHint | Qt::X11BypassWindowManagerHint);
 
             if (!hGrp->GetBool("ShowSplasherMessages", false)) {
                 d->splashscreen->setShowMessages(false);
             }
 
+            QEventLoop loop;
+            Q_UNUSED(loop)
             d->splashscreen->show();
+            QApplication::processEvents();
```

Comment:
What about:
```
            QEventLoop loop;
            d->splashscreen->show();
            loop.processEvents(); 
```

Reply:
Does it make any difference on your side?